Dockerfile

```diff
@@ -23,9 +23,26 @@ RUN set -ex; \
     sh /tmp/rustup.sh -y --no-modify-path --default-toolchain "$TOOLCHAIN"; \
     rm /tmp/*.sh
 
-COPY . /iroha/
+FROM rust-base as cargo-chef
+RUN cargo install cargo-chef
+
+FROM cargo-chef as planner
+WORKDIR /iroha
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
+
+ARG PROFILE=--release
```

Comment:
Maybe the default should be empty then then? So that we don't have to specify this argument in compose, and only specify it explicitly as `--release` in case it is needed.